pandas/tests/strings/test_get_dummies.py

```diff
@@ -22,19 +24,28 @@
 def test_get_dummies(any_string_dtype):
     s = Series(["a|b", "a|c", np.nan], dtype=any_string_dtype)
     result = s.str.get_dummies("|")
-    expected = DataFrame([[1, 1, 0], [1, 0, 1], [0, 0, 0]], columns=list("abc"))
+    exp_dtype = (
+        "boolean"
+        if any_string_dtype == "string" and any_string_dtype.na_value is NA
+        else "bool"
+    )
+    expected = DataFrame(
+        [[1, 1, 0], [1, 0, 1], [0, 0, 0]], columns=list("abc"), dtype=exp_dtype
+    )
     tm.assert_frame_equal(result, expected)
 
     s = Series(["a;b", "a", 7], dtype=any_string_dtype)
     result = s.str.get_dummies(";")
-    expected = DataFrame([[0, 1, 1], [0, 1, 0], [1, 0, 0]], columns=list("7ab"))
+    expected = DataFrame(
+        [[0, 1, 1], [0, 1, 0], [1, 0, 0]], columns=list("7ab"), dtype=exp_dtype
+    )
     tm.assert_frame_equal(result, expected)
 
 
 def test_get_dummies_index():
     # GH9980, GH8028
     idx = Index(["a|b", "a|c", "b|c"])
-    result = idx.str.get_dummies("|")
+    result = idx.str.get_dummies("|", dtype=np.int64)
```

Comment:
The behavior where the output becomes a `MultiIndex` when the input data is a `pd.Index` assumes that the `dtype` is not a boolean type:  
https://github.com/pandas-dev/pandas/blob/main/pandas/core/strings/accessor.py#L381-L389  

With this PR, the default behavior of `str.get_dummies()` changes to use a boolean dtype. To ensure the test cases remain consistent with the intended behavior, I modified them to explicitly specify the `dtype`.